src/java/com/unifina/data/Event.java

```diff
@@ -0,0 +1,57 @@
+package com.unifina.data;
+
+import java.util.Date;
+import java.util.function.Consumer;
+
+/**
+ * An Event is a wrapper class used for async dispatch of content to an attached Consumer.
+ * Event have a natural ordering based on the (timestamp, sequenceNumber) pair.
+ * @param <ContentClass> This is the type of the content, which is arbitrary. The generic is there to ensure that the content and Consumer are compatible.
+ */
+public class Event<ContentClass> implements Comparable<Event<ContentClass>> {
```

Comment:
The use of generics enforces the compatibility of constructor arguments `content` and `consumer`:

```
public Event(ContentClass content, Date timestamp, Consumer<ContentClass> consumer) {
```

This e.g. allows compile-time type checking in `Event` construction statements:

```
new Event<>(new MagnificentItem(), new Date(0), (content) -> {
    // We have compile-time knowledge that content is a MagnificentItem
    doSomethingWithMagnificentItem(content);
});
```

instead of (without generics):

```
new Event(new MagnificentItem(), new Date(0), (content) -> {
    // All we know is that content is an Object
    doSomethingWithMagnificentItem((MagnificentItem) content); // unsafe cast from compiler point of view
});
```